backend/models/project.js

```diff
@@ -18,266 +18,204 @@
-				projects.forEach(p => Project.populateUsers(userList, p));
-			}
+	Project.delete = async function(teamspace, name) {
+		const ModelHelper = require("./helper/model");
+		const project = await Project.findOneProject(teamspace, {name});
 
-			return projects;
-		});
+		if (!project) {
+			throw responseCodes.PROJECT_NOT_FOUND;
+		}
+
+		await db.remove(teamspace, PROJECTS_COLLECTION_NAME, {name});
+		// remove all models as well
+		if (project.models) {
+			await Promise.all(project.models.map(m => ModelHelper.removeModel(teamspace, m, true)));
+		}
+
+		return project;
 	};
 
-	schema.statics.findOneAndPopulateUsers = function(account, query) {
+	Project.listProjects = async function(teamspace, query = {}, projection) {
 		const User = require("./user");
-		let userList;
-
-		return User.getAllUsersInTeamspace(account.account).then(users => {
-			userList = users;
-			return Project.findOne(account, query);
-		}).then(project => {
-			if (project) {
-				return Project.populateUsers(userList, project);
-			} else {
-				return Promise.reject(responseCodes.PROJECT_NOT_FOUND);
-			}
-		});
-	};
+		const userList = await User.getAllUsersInTeamspace(teamspace);
+		const projects = await db.find(teamspace, PROJECTS_COLLECTION_NAME, query, projection);
 
-	schema.statics.findByNames = function(account, projectNames) {
-		return Project.find({account}, { name: { $in:projectNames } });
-	};
+		if (projects) {
+			projects.forEach(p => {
+				prepareProject(p);
+				populateUsers(userList, p);
+			});
+		}
 
-	schema.statics.findByIds = function(account, _ids) {
-		return Project.find({account}, { _id: { $in: _ids.map(utils.stringToUUID) } });
+		return projects;
 	};
 
-	schema.statics.populateUsers = function(userList, project) {
-		userList.forEach(user => {
-			const userFound = project.permissions.find(perm => perm.user === user);
+	Project.findProjectAndPopulateUsers = async function(teamspace, projectName) {
+		const User = require("./user");
+		const userList = await User.getAllUsersInTeamspace(teamspace);
+		const project = await Project.findOneProject(teamspace, {name: projectName});
 
-			if(!userFound) {
-				project.permissions.push({
-					user
-				});
-			}
-		});
+		if (!project) {
+			throw responseCodes.PROJECT_NOT_FOUND;
+		}
 
-		return project;
+		return populateUsers(userList, project);
 	};
 
-	schema.methods.findPermsByUser = function(username) {
-		return this.permissions.find(perm => perm.user === username);
-	};
+	Project.findProjectsById = async function(teamspace, ids) {
+		const foundProjects = await Project.listProjects(teamspace, { _id: { $in: ids.map(utils.stringToUUID) } });
 
-	schema.statics.isProjectAdmin = async function(account, model, user) {
-		const projection = { "permissions": { "$elemMatch": { user: user } }};
-		const project = await Project.findOne({account}, {models: model}, projection);
-		const hasProjectPermissions = project && project.permissions.length > 0;
+		foundProjects.forEach((project) => {
```

Comment:
there should be a clean function to do this? so it's inline with every other file?